src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -7429,192 +7429,199 @@ address generate_avx_ghash_processBlocks() {
   RuntimeStub* generate_cont_doYield() {
     if (!Continuations::enabled()) return nullptr;
 
-    const char *name = "cont_doYield";
-
     enum layout {
       rbp_off,
       rbpH_off,
       return_off,
       return_off2,
       framesize // inclusive of return address
     };
-    // assert(is_even(framesize/2), "sp not 16-byte aligned");
 
-    int insts_size = 512;
-    int locs_size  = 64;
-    CodeBuffer code(name, insts_size, locs_size);
-    OopMapSet* oop_maps  = new OopMapSet();
-    MacroAssembler* masm = new MacroAssembler(&code);
-    MacroAssembler* _masm = masm;
+    CodeBuffer code("cont_doYield", 512, 64);
+    MacroAssembler* _masm = new MacroAssembler(&code);
 
     address start = __ pc();
-
     __ enter();
-
-    __ movptr(c_rarg1, rsp);
-
-    int frame_complete = __ pc() - start;
     address the_pc = __ pc();
 
-    __ post_call_nop(); // this must be exactly after the pc value that is pushed into the frame info, we use this nop for fast CodeBlob lookup
+    int frame_complete = the_pc - start;
+
+    // This nop must be exactly at the PC we push into the frame info.
+    // We use this nop for fast CodeBlob lookup, associate the OopMap
+    // with it right away.
+    __ post_call_nop();
+    OopMapSet* oop_maps = new OopMapSet();
+    OopMap* map = new OopMap(framesize, 1);
+    oop_maps->add_gc_map(the_pc - start, map);
 
-    __ movptr(c_rarg0, r15_thread);
     __ set_last_Java_frame(rsp, rbp, the_pc);
+    __ movptr(c_rarg0, r15_thread);
+    __ movptr(c_rarg1, rsp);
     __ call_VM_leaf(Continuation::freeze_entry(), 2);
```

Comment:
Was it bug to not set `c_rarg1`?  stubGenerator_aarch64.cpp has the same issue.

Reply:
Ah, so this one is little "behavioral" change. For some weird reason, the `__ movptr(c_rarg1, rsp);` used to be much earlier, between `start` and `the_pc`. I don't believe that's part of the frame setup. It looks like a copy-paste glitch from `generate_throw_exception`. So I just moved it here. No tests complain about this move.

Reply:
Okay. I missed it in original place (too far away) and I thought it is new instruction.